spec/v1.0/statement.md

```diff
@@ -33,37 +33,24 @@ Additional [parsing rules] apply.
 > Identifier for the schema of the Statement. Always
 > `https://in-toto.io/Statement/v1` for this version of the spec.
 
-`subject` _array of objects, required_
+`subject` _array of [ResourceDescriptor] objects, required_
 
 > Set of software artifacts that the attestation applies to. Each element
-> represents a single software artifact.
+> represents a single software artifact. Each object MUST have `digest` set.
+> The `name` field may be used as an identifier to distinguish this artifact
+> from others within the `subject`. The semantics are up to the producer and
+> consumer and they MAY use it when evaluating policy. If the name is not
+> meaningful, leave the field unset or use "\_". For example, a
+> [SLSA Provenance] attestation might use the name to specify output filename,
+> expecting the consumer to only consider entries with a particular name.
+> Alternatively, a vulnerability scan attestation might leave name unset because
+> the results apply regardless of what the artifact is named. If set, `name`
+> SHOULD be unique within subject.
 >
 > IMPORTANT: Subject artifacts are matched purely by digest, regardless of
 > content type. If this matters to you, please comment on
 > [GitHub Issue #28](https://github.com/in-toto/attestation/issues/28)
 
-`subject[*].digest` _object ([DigestSet]), required_
-
-> Collection of cryptographic digests for the contents of this artifact.
->
-> Two DigestSets are considered matching if ANY of the fields match. The
-> producer and consumer must agree on acceptable algorithms. If there are no
-> overlapping algorithms, the subject is considered not matching.
```

Comment:
We need to describe and maintain the semantics of what an attestation is bound to. Currently it is definitely the `digest` (using the matching rules described here) and optionally `name` if the producer and consumer agree on it. Since we're adding more fields, I think basically we'd be saying that it MUST match on `digest` and MAY match on other fields if the producer and consumer agree on that?

Reply:
I don't know that the rule belongs here actually. We have the matching rules for digest objects in the `DigestSet` definition. As for how to evaluate if the resource descriptor object as a whole should be matched, it's perhaps context specific? In some cases we may be matching on name and allowing digest to change (eg: `MODIFY foo/*`).

Reply:
Oops, I missed that and also the "subject artifacts are matched purely by digest". Please disregard this comment.